DIP/DIP/Models/CardPaymentService.cs

```diff
@@ -0,0 +1,22 @@
+﻿using DIP.Interfaces;
+
+namespace DIP
+{
+    public class CardPaymentService<CardPayment> : IPaymentService<CardPayment> where CardPayment : IPayment
```

Comment:
Card payment service class does not need to be generic.
You don't have to add where condition to CardPayment for two reasons
1- you already have the condition in IPaymentService, this should be enough to make sure you don't pass any other type.
2- You're using specific class (CardPayment) not generic type (T) like you used in the interface.
you're code should look like this:
public class CardPaymentService : IPaymentService<CardPayment>

let me know if you have any question regarding this